pkg/controller/elasticsearch/observer/manager.go

```diff
@@ -7,47 +7,61 @@ package observer
 import (
 	"sync"
 
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/annotation"
 	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/client"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"go.elastic.co/apm"
 	"k8s.io/apimachinery/pkg/types"
 )
 
+const (
+	// ObserverIntervalAnnotation is the name of the annotation used to set the observation interval for a cluster.
+	ObserverIntervalAnnotation = "eck.k8s.elastic.co/es-observer-interval"
+	// ObserverRequestTimeoutAnnotation is the name of the annotation used to set the observe request timeout for a cluster.
+	ObserverRequestTimeoutAnnotation = "eck.k8s.elastic.co/es-observer-request-timeout"
+)
+
 // Manager for a set of observers
 type Manager struct {
 	observers map[types.NamespacedName]*Observer
 	listeners []OnObservation // invoked on each observation event
 	lock      sync.RWMutex
-	settings  Settings
+	tracer    *apm.Tracer
 }
 
 // NewManager returns a new manager
-func NewManager(settings Settings) *Manager {
+func NewManager(tracer *apm.Tracer) *Manager {
 	return &Manager{
 		observers: make(map[types.NamespacedName]*Observer),
 		lock:      sync.RWMutex{},
-		settings:  settings,
+		tracer:    tracer,
 	}
 }
 
 // ObservedStateResolver returns the last known state of the given cluster,
 // as expected by the main reconciliation driver
-func (m *Manager) ObservedStateResolver(cluster types.NamespacedName, esClient client.Client) State {
+func (m *Manager) ObservedStateResolver(cluster *esv1.Elasticsearch, esClient client.Client) State {
```

Comment:
nit: do we need a pointer here? Not passing pointers would give a better hint that this cluster object won't be mutated (even though technically you could also mutate arrays of a non-pointer but well...). So I tend to prefer not using pointers unless there are strictly required.
Same for `Observe()` and `extractObserverSettings()`.

Reply:
I wanted to avoid a potentially large copy of the Elasticsearch object; but, you're right, it probably doesn't matter here.